### PR TITLE
Fix valid timezone designators not being accepted.

### PR DESCRIPTION
**Pull request by chrisgavin**
According to the [JSON schema specification, "date-time" formatted strings should be represented in ISO8601 format](https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times). [ISO8601 allows time zone offsets of the form ±hh:mm, ±hhmm and ±hh](https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC), but currently only the first of those formats is supported.

This pull request updates the regex for `date-time`s so that the latter two timezone formats are also accepted.

### CODE REVIEW

File: lib/json_schema/validator.rb [hunk 1]
```diff
@@ -593,7 +593,7 @@ def find_parent(schema)
 
     DATE_PATTERN = /^[0-9]{4}-[0-9]{2}-[0-9]{2}$/
 
-    DATE_TIME_PATTERN = /^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-2][0-9]:[0-5][0-9]:[0-5][0-9](\.[0-9]+)?(Z|[\-+][0-9]{2}:[0-5][0-9])$/
+    DATE_TIME_PATTERN = /^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-2][0-9]:[0-5][0-9]:[0-5][0-9](\.[0-9]+)?(Z|[\-+][0-9]{2}(:?[0-5][0-9])?)$/
```
Comment (aibaars) [inline lib/json_schema/validator.rb:596]: As far as I know `Z` is also a valid timezone designator, best add that too.

Comment (aibaars) [reply lib/json_schema/validator.rb:596]: Ah scratch that, I didn't spot the `Z` in this long regex.

**Review by aibaars — CHANGES_REQUESTED** [review]
I'm not sure this pull request is correct. The specification states:
```
Dates and times are represented in RFC 3339, section 5.6. This is a subset of the date format also commonly known as ISO8601 format.
```
`RFC 3339, section 5.6` is a subset of ISO8601, and therefore not all ISO8601 date-times should be considered valid.

Section 5.6 of https://tools.ietf.org/html/rfc3339#section-5.6 states:
```
5.6. Internet Date/Time Format

   The following profile of ISO 8601 [ISO8601] dates SHOULD be used in
   new protocols on the Internet.  This is specified using the syntax
   description notation defined in [ABNF].

   date-fullyear   = 4DIGIT
   date-month      = 2DIGIT  ; 01-12
   date-mday       = 2DIGIT  ; 01-28, 01-29, 01-30, 01-31 based on
                             ; month/year
   time-hour       = 2DIGIT  ; 00-23
   time-minute     = 2DIGIT  ; 00-59
   time-second     = 2DIGIT  ; 00-58, 00-59, 00-60 based on leap second
                             ; rules
   time-secfrac    = "." 1*DIGIT
   time-numoffset  = ("+" / "-") time-hour ":" time-minute
   time-offset     = "Z" / time-numoffset

   partial-time    = time-hour ":" time-minute ":" time-second
                     [time-secfrac]
   full-date       = date-fullyear "-" date-month "-" date-mday
   full-time       = partial-time time-offset

   date-time       = full-date "T" full-time

      NOTE: Per [ABNF] and ISO8601, the "T" and "Z" characters in this
      syntax may alternatively be lower case "t" or "z" respectively.

      This date/time format may be used in some environments or contexts
      that distinguish between the upper- and lower-case letters 'A'-'Z'
      and 'a'-'z' (e.g. XML).  Specifications that use this format in
      such environments MAY further limit the date/time syntax so that
      the letters 'T' and 'Z' used in the date/time syntax must always
      be upper case.  Applications that generate this format SHOULD use
      upper case letters.

      NOTE: ISO 8601 defines date and time separated by "T".
      Applications using this syntax may choose, for the sake of
      readability, to specify a full-date and full-time separated by
      (say) a space character.
```
I understand the above to mean  `hh` and `mm` are mandatory and the `:` as separator between them is mandatory too.

**Comment by chrisgavin** [comment]
Ah apologies @aibaars. I had not realized that only a subset of ISO8601 was supported!

**Pull request closed without merging**